lib/ex_force/request.ex

```diff
@@ -0,0 +1,24 @@
+defmodule ExForce.Request do
+  @moduledoc """
+  Represents an ExForce API request.
+  """
+
+  @type param() :: binary() | [{binary() | atom(), param()}]
+  @type t :: %__MODULE__{
+          body: any(),
+          headers: [{binary(), binary()}],
+          method: :head | :get | :delete | :trace | :options | :post | :put | :patch,
+          opts: [any()],
+          query: [{binary() | atom(), param()}],
```

Comment:
Hm... does tesla support nested params here? If we don't use any nested params, let's just make it flat query (not including `param` inside `query`)

By nested params I mean:

```elixir
[parent: [child1: val1, child2: val2]]
```

Reply:
Yes, [tesla supports nested params](https://hexdocs.pm/tesla/Tesla.Env.html#t:query/0), but I agree, let's keep it flat.

Implemented in 64ad54e108e733b617ceba9701347e181e2eacce